trigger/patch.go

```diff
@@ -118,6 +118,49 @@ func (t *patchTriggers) patchOutcome(sub *event.Subscription) (*notification.Not
 		}
 	}
 
+	if t.patch.IsParent() || (t.patch.IsChild() && sub.Subscriber.Type == event.ParentWaitOnChildSubscriberType) {
+		// get the children or siblings to wait on
+		childrenOrSiblings, parentPatch, err := t.patch.GetPatchFamily()
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting child or sibling patches")
+		}
+
+		childrenStatus, unreadyChild, err := getChildrenOrSiblingsReadiness(childrenOrSiblings)
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting child or sibling information")
+		}
+		//make sure the children or siblings are done before sending the notification
+		if !evergreen.IsFinishedPatchStatus(childrenStatus) {
+			parentId := t.patch.Id.Hex()
+			if t.patch.IsChild() {
+				parentId = parentPatch.Id.Hex()
+			}
+			// if there is still a child or sibling that's not done, subscribe on it and don't create the notification
+			err = subscribeOnChild(parentId, unreadyChild.Id.Hex(), sub)
+			if err != nil {
+				return nil, errors.Wrap(err, "error subscribing on child patch")
+			}
+			return nil, nil
+		}
+
+		if childrenStatus == evergreen.PatchFailed {
+			t.data.Status = evergreen.PatchFailed
+		}
+
+		// convert the subscription to the right type
+		if sub.Subscriber.Type == event.ParentWaitOnChildSubscriberType {
+			t.patch = parentPatch
+			target, ok := sub.Subscriber.Target.(*event.ParentWaitOnChildSubscriber)
+			if !ok {
+				return nil, errors.Errorf("target '%s' didn't not have expected type", sub.Subscriber.Target)
+			}
+			target.OriginalSub.LastUpdated = time.Now()
```

Comment:
I'm recycling the old subscriber here. If a subscription comes in for a parent with unfinished children, we subscribe on the child with the original subscription saved. Then, when the child patch is done, we grab the original subscription, update the time to be the new time instead of the original time, and generate for the original subscription on the parent because we are now ready for it.

Reply:
With the suggestion to create subscriptions for all children immediately, I think you can have the same subscriber for all of them and don't need this block

Reply:
fixed, thanks!